hs_core/hydroshare/resource.py

```diff
@@ -686,7 +690,18 @@ def delete_resource(pk):
 
     Note:  Only HydroShare administrators will be able to delete formally published resour
     """
-    utils.get_resource_by_shortkey(pk).delete()
+    #utils.get_resource_by_shortkey(pk).delete()
+    res = utils.get_resource_by_shortkey(pk)
+    for fl in res.files.all():
+        fl.resource_file.delete()
+
+    for bag in res.bags.all():
+        bag.bag.delete()
+        bag.delete()
+
+    res.metadata.delete_all_elements()
+    # deleting the metadata object deletes the resource also
+    res.metadata.delete()
```

Comment:
I think you should put this back into the `delete()` method of the resource model. If the model instance is deleted, it should know to clean itself up. This will ensure that cleanup always happens  from any place a resource gets deleted.

I have some separate concerns about the safety of this delete, in terms of a failure half-way through. Because of the mix of Database and File System bits of a resource, there isn't a simple way to make the delete atomic. I think an issue for that should be filed to address in the future.